### PR TITLE
[Feature][CDC] MySQL CDC supports deserialization of multi-tables

### LINKED ISSUE #4051: [Feature][CDC] MySQL CDC supports deserialization of multi-tables
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

This is a subtask of https://github.com/apache/incubator-seatunnel/issues/4050 to track completion.

1. It's TableSourceFactory implement the SupportMultipleTable interface, use the information of multiple CatalogTables to create MultipleRowType, and SeaTunnelSource#getProducedType will return MultipleRowType.
2. Use MultipleRowType inside Source to deserialize data into SeaTunnelRow, and add table name to SeaTunnelRow.

Make MySQL CDC support the above features.

#### Related pseudo-code

```java
// For Source deserialization and Row distribution
public class MultipleRowType implements SeaTunnelDataType<SeaTunnelRow> {
    private final String[] tableNames;
    private final SeaTunnelRowType[] rowTypes;
}
```

```java
// Declare that the Source supports multiple tables, and control the number of tables by itself
public interface SupportMultipleTable {
    /**
     * A connector can pick tables and return the accepted and remaining tables.
     */
    Result applyTables(TableFactoryContext context);

    final class Result {
        private final List<CatalogTable> acceptedTables;
        private final List<CatalogTable> remainingTables;

        private Result(
                List<CatalogTable> acceptedTables,
                List<CatalogTable> remainingTables) {
            this.acceptedTables = acceptedTables;
            this.remainingTables = remainingTables;
        }
    }
}
```

### Usage Scenario

MySQL CDC

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hailin0**
## Purpose of this pull request

close #4051

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
@lizong
PTAL

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java [hunk 1]
```diff
@@ -76,6 +79,14 @@
     protected StopMode stopMode;
     protected DebeziumDeserializationSchema<T> deserializationSchema;
 
+    protected SeaTunnelDataType<SeaTunnelRow> dataType;
+    protected ReadonlyConfig options;
+
+    protected IncrementalSource(ReadonlyConfig options, SeaTunnelDataType<SeaTunnelRow> dataType) {
+        this.options = options;
+        this.dataType = dataType;
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/IncrementalSource.java]: The `#prepare` method will not be called in the new process, please initialize the remaining `fields`.

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializeSchema.java [hunk 2]
```diff
@@ -90,28 +114,34 @@ public void deserialize(SourceRecord record, Collector<SeaTunnelRow> collector)
         Schema valueSchema = record.valueSchema();
 
         Struct sourceStruct = messageStruct.getStruct(Envelope.FieldName.SOURCE);
-        // TODO: multi-table
+        String database = sourceStruct.getString(DATABASE_NAME_KEY);
         String tableName = sourceStruct.getString(AbstractSourceInfo.TABLE_NAME_KEY);
+        String tableId = database + ":" + tableName;
+        SeaTunnelRowDebeziumDeserializationConverters converters = multipleRowConverters.getOrDefault(tableId, singleRowConverter);
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializeSchema.java:120]: The judgment on the single table is missing

Comment (hailin0) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializeSchema.java:120]: through `multipleRowConverters.getOrDefault(..., singleRowConverter)`

**Review by ashulin — APPROVED** [review]
LGTM

**Pull request merged**